test/test_response.py

```diff
@@ -377,6 +377,59 @@ def test_empty_stream(self):
 
         self.assertRaises(StopIteration, next, stream)
 
+    def test_length_no_header(self):
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, preload_content=False)
+        assert resp.length is None
+
+    def test_length_w_valid_header(self):
+        headers = {"content-length": "5"}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length == 5
+
+    def test_length_w_bad_header(self):
+        garbage = {'content-length': 'foo'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+        garbage['content-length'] = "-10"
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+    def test_length_when_chunked(self):
+        headers = {'content-length': '5',
+                   'transfer-encoding': 'chunked'}
```

Comment:
Yeah, this boils down to "how do we want to deal with this"? The options are as follows:

1. Raise an exception explaining what went wrong. This fails fast and clearly.
2. Fall back to no content-length. That means we'll treat the body as chunked. If it's not, we'll fail fast (IncompleteRead, usually). If it is, everything works.

I think that's probably ok in this case, but I'd like a comment explaining the rationale.